profiles/commands.py

```diff
@@ -15,6 +20,38 @@ def _update_name_from_orcid_record(profile: Profile, orcid_record: dict) -> None
                             '{}, {}'.format(family_name, given_name))
 
 
+def _update_affiliations_from_orcid_record(profile: Profile, orcid_record: dict) -> None:
+    orcid_affiliations = orcid_record.get('activities-summary', {}).get('employments', {}).get(
+        'employment-summary', {})
+
+    for index, orcid_affiliation in enumerate(orcid_affiliations):
+        organization = orcid_affiliation['organization']
+        address = organization['address']
+
+        affiliation = Affiliation(str(orcid_affiliation['put-code']),
+                                  countries.get(address['country']), organization['name'],
+                                  date(**{k: int(v['value']) for k, v in
+                                          orcid_affiliation['start-date'].items()}),
+                                  orcid_affiliation.get('department-name'), address.get('city'),
+                                  address.get('region'),
+                                  date(**{k: int(v['value']) for k, v in
+                                          orcid_affiliation['end-date'].items()}, hour=23,
+                                       minute=59, second=59) if orcid_affiliation.get('end-date')
+                                  else None,
+                                  orcid_affiliation['visibility'] != VISIBILITY_PUBLIC)
+        profile.add_affiliation(affiliation, index)
+
+    for affiliation in profile.affiliations:
```

Comment:
it looks like this iterates on `profile.affiliations` to diff it from `orcid_affiliations`, but two lines ago everything from `orcid_affiliatons` was added to `profile`?

Reply:
To find any entries that have been removed.

Reply:
I see how that works now, but if we build a `added_ids` `Set` the complexity of this can be reduced to `if affiliation.id in added_ids` (both in terms of computation and readability)

Reply:
👍